lib/dcv/solr/document_adapter/active_fedora.rb

```diff
@@ -0,0 +1,213 @@
+        else
+          solr_doc["contributor_first_si"] = solr_doc["contributor_ssim"]
+        end
+      end
+
+      solr_doc["format_ssi"] = [route_as]
+
+      set_size_labels(solr_doc)
+
+      solr_doc.each_pair do |key, value|
+        if value.is_a? Array
+          value.each {|v| v.strip! unless v.nil? }
+        elsif value.is_a? String
+          value.strip!
+        end
+      end
+
+      solr_doc['structured_bsi'] = true if has_struct_metadata?
+
+      get_representative_generic_resource&.tap { |rgr| solr_doc['representative_generic_resource_pid_ssi'] = rgr.pid }
+
+      # Index URI form of pid to facilitate solr joins
+      solr_doc['fedora_pid_uri_ssi'] = 'info:fedora/' + obj.pid if obj.pid.present?
+      solr_doc['datastreams_ssim'] = obj.datastreams.keys.map {|k| k.to_s }.sort
+
+      # add XACML indexing
+      if has_access_control_metadata?
+        Dcv::Solr::DocumentAdapter::XacmlXml.new(obj.datastreams['accessControlMetadata'].content).to_solr(solr_doc)
+      end
+
+      # add RELS-INT indexing
+      if has_rels_int?
+        rels_int.to_solr(solr_doc)
+      end
+
+      solr_doc['all_text_teim']&.uniq!
+
+      solr_doc
+    end
+
+    def route_as
+      'default'
+    end
+
+    def index_type_label
+      'DEFAULT'
+    end
+
+    # set the index type label and any RI-based fields
+    def set_size_labels(solr_doc={})
+      solr_doc["index_type_label_ssi"] = [index_type_label]
+    end
+
+    # legacy behavior is more stringent than other ds methods
+    def has_desc?
+      if obj.datastreams['descMetadata']&.has_content?
+        return Nokogiri::XML(obj.datastreams['descMetadata'].content).xpath('/mods:mods/mods:identifier', MODS_NS).first
+      end
+      false
+    end
+
+    def has_struct_metadata?
+      return obj.datastreams['structMetadata']&.has_content?
+    end
+
+    def has_access_control_metadata?
+      return obj.datastreams['accessControlMetadata']&.has_content?
+    end
+
+    def has_rels_int?
+      return obj.datastreams['RELS-INT']&.has_content?
+    end
+
+    def rels_int
+      content = obj.datastreams["RELS-INT"].content
+      ds = obj.create_datastream(Cul::Hydra::Datastreams::RelsInt, "RELS-INT", controlGroup: 'X', blob: content)
+      ds
+    end
+
+    def get_singular_relationship_value(predicate)
+      get_relationship_values(predicate)&.first
+    end
+
+    def self.get_relationship_values(obj, predicate)
+      properties = obj.relationships(predicate)
+      return [] unless properties.present?
+      return properties.map { |property| (property.kind_of? RDF::Literal) ? property.value : property }
+    end
+
+    def get_relationship_values(predicate)
+      ActiveFedora.get_relationship_values(obj, predicate)
+    end
+
+    # determine whether an object asserts any of the specified cmodels
+    # test_cmodels(["info:fedora/ldpd:GenericResource"])
+    # @params [Array<String>] cmodels
+    def self.test_cmodels(obj, cmodels)
```

Comment:
Optional suggestion, but renaming this method to something like `matches_any_cmodel?` may be slightly clearer, since this check reminds me a bit of `is_a?` and I didn't initially guess that behavior (plus boolean return value) from the `test_cmodels` name.